currentMonitoring.cpp

```diff
@@ -0,0 +1,46 @@
+#include "currentMonitoring.h"
+#include <stdio.h>
+#include <vector>
+#include <algorithm>
+#include <cmath>
+
+std::vector<int> Icurrentreadings::sortCurrentSamples(std::vector<int>& samples) {
+	sort(samples.begin(), samples.end());
+	return samples;
+}
+std::vector<int> Icurrentreadings::removeDuplicateFromCurrentSamples(std::vector<int>& samplesSamples) {
+
+	samplesSamples.erase(unique(samplesSamples.begin(), samplesSamples.end()), samplesSamples.end());
+	return samplesSamples;
+}
+std::map<int, int> Icurrentreadings::countFrequency(std::vector<int>& samples) {
+	for (auto const & sample : samples) {
+		int freq = std::count(samples.begin(), samples.end(), sample);
+		if (freq >= 2) {
+			frequency[sample] = freq;
+		}
+	}
+	return frequency;
+}
+int Icurrentreadings::readingFromRange(std::vector<int>& checkReadings, std::map<int, int > frequecyHandler) {
+	int temp = checkReadings.size();
+	for(auto const & freq : frequecyHandler) {
+		if (std::find(checkReadings.begin(), checkReadings.end(), freq.first) != checkReadings.end()) {
+			temp = (checkReadings.size() - 1) + freq.second;
+		}
+	}
+	return temp;
+}
+void Icurrentreadings::detectRange(std::vector<int>& continuousReadings, std::map<int, int > frequecyHandler) {

```

Comment:
This `detectRange` does not have a test. Remove it, write the test and implement. Let's see if the implementation changes

Reply:
i will update